Updates Android & iOS release lanes to distribute beta build to external users

ggirotto opened the pull request:
Introduces a performance improvement to release workflow that may save around 5 minutes, according to the last release workflow logs:

![CleanShot 2021-07-16 at 11 10 12@2x](https://user-images.githubusercontent.com/11745745/125961719-0adeb4d1-05df-42fe-b4ca-07217243f29c.png)

It basically ignores the processing time from AppStoreConnect. It doesn't make sense to wait for it since we'll need to manually release it latter in the portal.

matuella reviewed (commented):
Didn't `distribute_external: true` automatically upload to all external testers?

ggirotto commented:
@matuella Yes, but I think we'll need to provide the correctly CHANGELOG to them anyway. I though about using `Changelog.txt` that I removed in this PR, but I don't think this is going to work unless we keep it updated as we updated the base CHANGELOG.md. What do you think?

matuella commented:
> @matuella Yes, but I think we'll need to provide the correctly CHANGELOG to them anyway. I though about using `Changelog.txt` that I removed in this PR, but I don't think this is going to work unless we keep it updated as we updated the base CHANGELOG.md. What do you think?

I think the `CHANGELOG.md` is too technical for Testflight testers. I think these are good to support, although we will have to update these alongside `CHANGELOG.md`.

On a side note, how Play Store deals with version changelogs (for testers)?

ggirotto commented:
We can have a similar structure in the Android target. [Fastlane Supply](https://docs.fastlane.tools/actions/supply/) supports CHANGELOG txt files as we are doing in iOS today. Do you think we should keep the CHANGELOG and automatically distribute new release to external testers or manually release these versions?

matuella commented:
> We can have a similar structure in the Android target. [Fastlane Supply](https://docs.fastlane.tools/actions/supply/) supports CHANGELOG txt files as we are doing in iOS today. Do you think we should keep the CHANGELOG and automatically distribute new release to external testers or manually release these versions?

I think automatic distributing for releases are nice, honestly - this means that store releases will still required a manual effort - as expected. Given that the work is pretty much summarizing the release in a .txt file, I don't see why not do it.

ggirotto commented:
@matuella Updated this PR to correctly distribute beta builds with their respective CHANGELOG to iOS and Android stores
